src/app/public/modules/i18n/currency/currency-format.service.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  getCurrencySymbol
+} from '@angular/common';
+
+import {
+  SkyCurrencyFormat
+} from './currency-format';
+
+/**
+ * Service to get a currency's format given an iso code.
+ */
+ @Injectable({
+  providedIn: 'root'
+})
+export class SkyCurrencyFormatService {
+
+  /**
+   * Gets a currency's format.
+   * @param params optional params object
+   * @param params.isoCurrencyCode the ISO 4217 Currency Code. Defaults to `USD`.
+   * @param params.locale the locale. Defaults to `en-US`. Examples: `en-US`, `en-GB`, `fr-FR`.
+   */
+  public getCurrencyFormat(params?: Partial<IsoCurrencyCodeAndLocale>): SkyCurrencyFormat {
+    const isoCurrencyCode = params?.isoCurrencyCode ?? 'USD';
+    const locale = params?.locale ?? 'en-US';
+    const formatter = new Intl.NumberFormat(locale, {
+      style: 'currency',
+      currency: isoCurrencyCode
+    });
+
+    const resolvedOptions: Intl.ResolvedNumberFormatOptions = formatter.resolvedOptions();
+    const currencyCode = resolvedOptions.currency;
+    const parts = this.formatToParts(formatter, currencyCode, locale);
+
+    return {
+      locale: locale,
+      isoCurrencyCode: currencyCode,
+      symbol: parts.symbol,
+      symbolLocation: parts.symbolLocation,
+      decimalCharacter: parts.decimalCharacter,
+      groupCharacter: parts.groupCharacter,
+      precision: resolvedOptions.maximumFractionDigits
+    };
+  }
+
+  /**
+   * Given an ISO 4217 Currency Code return the currency's precision.
+   * @param isoCurrencyCode the ISO 4217. Defaults to `USD`.
+   */
+  public getCurrencyPrecision(isoCurrencyCode: string): number {
```

Comment:
`getCurrencyPrecision`` doesn't _need_ to be but since users could call `SkyCurrencyFormatService.getCurrencyFormat` and grab the precision from the object. 

Its definitely a convenience method for when you want JUST a currency's precision and it helps drive that different currencies can have different precisions other than USD's 2.